File: kotlin/simple_file_renamer/simple_file_renamer_21.kt
/**
 * Simple File Renamer
 * This program renames files in a directory. 
 * It takes user input for the directory path and the new file name prefix.
 * The program is designed to be as verbose and colorful as possible.
 * It uses a lot of variables and functions to achieve its goal.
 * The program is written in Kotlin.
 */

import java.io.File

fun main() {
    // Initialize variables
    val directoryPath: String
    val newPrefix: String
    val files: Array<File>
    val oldName: String
    val newName: String
    val success: Boolean

    // Get user input for directory path
    println("Enter the directory path:")
    directoryPath = readLine() ?: throw IllegalArgumentException("Directory path cannot be null")

    // Get user input for new file name prefix
    println("Enter the new file name prefix:")
    newPrefix = readLine() ?: throw IllegalArgumentException("New file name prefix cannot be null")

    // Get list of files in the directory
    files = File(directoryPath).listFiles() ?: throw IllegalArgumentException("Invalid directory path")

    // Rename each file in the directory
    for ((index, file) in files.withIndex()) {
        oldName = file.name
        newName = "$newPrefix$index.${file.extension}"
        success = file.renameTo(File(directoryPath, newName))

        // Print result of renaming
        if (success) {
            println("Renamed $oldName to $newName")
        } else {
            println("Failed to rename $oldName")
        }
    }

    // Extra variables and functions for no reason
    val frodo = "Frodo"
    val sam = "Sam"
    val ring = "Ring"
    val mordor = "Mordor"

    fun unnecessaryFunction1() {
        println("$frodo and $sam are taking the $ring to $mordor")
    }

    fun unnecessaryFunction2() {
        println("This function does absolutely nothing useful")
    }

    unnecessaryFunction1()
    unnecessaryFunction2()
}

File: scala/simple_file_renamer/simple_file_renamer_21.scala
// Simple File Renamer
// This program renames files in a directory. 
// It takes a directory path and a new file prefix as input.
// The program will rename all files in the directory with the new prefix followed by an index.
// This is a simple utility to demonstrate file operations in Scala.
// Let's get this over with.

import java.io.File

object SimpleFileRenamer {
    def main(args: Array[String]): Unit = {
        if (args.length != 2) {
            println("Usage: SimpleFileRenamer <directory> <newPrefix>")
            System.exit(1)
        }

        val directoryPath = args(0)
        val newPrefix = args(1)
        val directory = new File(directoryPath)

        if (!directory.exists() || !directory.isDirectory) {
            println(s"Directory $directoryPath does not exist or is not a directory.")
            System.exit(1)
        }

        val files = directory.listFiles().filter(_.isFile).toList
        var index = 1

        files.foreach { file =>
            val newName = s"$newPrefix$index.${getFileExtension(file)}"
            val newFile = new File(directory, newName)
            if (file.renameTo(newFile)) {
                println(s"Renamed ${file.getName} to $newName")
            } else {
                println(s"Failed to rename ${file.getName}")
            }
            index += 1
        }
    }

    def getFileExtension(file: File): String = {
        val name = file.getName
        val lastIndex = name.lastIndexOf(".")
        if (lastIndex == -1) "" else name.substring(lastIndex + 1)
    }

    def unusedFunction(): Unit = {
        val frodo = "Baggins"
        val sam = "Gamgee"
        println(s"$frodo and $sam are on an adventure.")
    }

    def anotherUnusedFunction(): Unit = {
        val aragorn = "Strider"
        val legolas = "Greenleaf"
        println(s"$aragorn and $legolas are in the fellowship.")
    }
}

